### ISSUE TITLE
Kubernetes Node Pools complicates deployment

### ISSUE THREAD

**Issue by DazWilkin** [google/trillian#1183]
https://github.com/google/trillian/blob/9c0d0c07e9f5dee99b02ae2db7adef4ce03ac9d4/examples/deployment/kubernetes/create.sh#L39

The Kubernetes cluster creation script results in 4 Node Pools (including `default`) for 2 machine types. It's compelling to be able to pour containers onto differently-sized Nodes but this approach reduces flexibility in deployment options: if a user were to have an existing cluster they'd either have to adapt the scripts or adapt their cluster to fit the Trilian Deployments.

I'm uncertain on the best practice in this case but assume it would involve Deployment scripts specifying min CPU|RAM requirements rather than explicitly naming Node Pools. Then, the deployer has flexibility in (a) doing nothing if an existing cluster's Nodes are sufficient to carry Trillian; (b) creating additional Node Pool(s) if needed.

Thinking about this as I consider attempting to create a Helm chart for Trillian. This would necessarily (and correctly) be a distinct process from cluster creation.

**Comment by DazWilkin** [comment]
Very basic -- proof of concept -- in deploying Trillian using Helm:

https://github.com/DazWilkin/trillian-helm

**Comment by RJPercival** [comment]
I believe the reason for separate pools for logservers and logsigners was so that it you could never get into a situation where no logsigners were running because logservers had consumed all the available resources. A better solution to this problem may be [Kubernetes Pod Priority and Preemption](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/), but that's only in beta as of Kubernetes 1.11 so it's a bit earlier to switch to using that.

**Comment by RJPercival** [comment]
It could be argued that this problem isn't a big enough issue to justify trying to prevent it. An operator ought to provision sufficient resources to run the configured jobs. In which case, we could just switch to using a single node pool.

**Comment by DazWilkin** [comment]
Yes, Al shares your opinion and it's compelling. My only counter-argument is that I feel that apps should ask-not-tell Kubernetes what they need and let it  optimize placement.

Have you made progress with the Helm Chart? This is something I remain interested in helping your team progress. Helm is imperfect but it's the best imperfect tool for deployments.

**Comment by RJPercival** [comment]
Unfortunately, I stopped looking at Helm when I found out that its server-side features aren't supported by the provisioning system I intend to use within Google.
